src/federated.c

```diff
@@ -108,14 +109,8 @@ void FederatedOutputConnection_cleanup(Trigger *trigger) {
 
     LF_DEBUG(FED, "FedOutConn %p sending tagged message with tag=%" PRId64 ":%" PRIu32, trigger, tagged_msg->tag.time,
              tagged_msg->tag.microstep);
-    ret = channel->send_blocking(channel, &msg);
-    switch (ret) {
-    case LF_OK:
-      break;
-    case LF_CONNECTION_CLOSED:
+    if (channel->send_blocking(channel, &msg) != LF_OK) {
       self->bundle->network_channel_state_changed(self->bundle);
```

Comment:
If we remove this "callback"-way of handling network-channel state updates (see my other comment), then this call should probably also be dropped. Instead we check the network channel state each time before going to sleep waiting for some input port to become known at a tag

Reply:
Regarding all the callback comments I think that change should go into the second PR. I tried to keep the scope of this one only within the deduplication of the return code and state :)